src/mesh/poly2tri_triangulator.C

```diff
@@ -1500,7 +1502,8 @@ bool Poly2TriTriangulator::should_refine_elem(Elem & elem)
   // vertices first
   for (auto v : make_range(elem.n_vertices()))
     {
-      const Real local_area_target = (*area_func)(elem.point(v));
+      // If we have an auto area function, we'll use it and override other area options
+      const Real local_area_target = this->has_auto_area_function() ? this->get_auto_desired_area(elem.point(v)) : (*area_func)(elem.point(v));
```

Comment:
It's still taking a `Point` as input and outputting a scalar as output, so couldn't we encapsulate the interpolation in a `FunctionBase` subclass rather than in a member function here?  The extra class definition would be a little bit more boilerplate but it would also eliminate half of this diff.